docs/source/tutorials/configurable.rst

```diff
@@ -0,0 +1,108 @@
+==============================
+Configuration and plugin setup
+==============================
+
+Maybe you :doc:`played with commands </tutorials/playing-command>` for your
+plugin and now you want to make your plugin configurable. If you run an
+instance of Sopel yourself, you probably had to open and edit its configuration
+file.
+
+Usually located in your home directory, under the ``.sopel/`` folder, the
+configuration file is an INI file with sections defined by Sopel's core and by
+plugins. In this tutorial, let's see how to declare and use a configuration
+section dedicated to your plugin.
+
+
+Declaring your configuration
+============================
+
+To declare a configuration section, you must first create a subclass of
+:class:`~sopel.config.types.StaticSection`, and define attributes::
```

Comment:
I think it would help to clarify what `define attributes` means here and allude to the fact that there are two different kinds of attributes at play: Python class attributes and Sopel's attribute types for settings. I tried not to get carried away in my suggestion.

```suggestion
To declare a configuration section, you create a subclass of
:class:`~sopel.config.types.StaticSection` that defines a class attribute for
each setting that can be configured in that section. To declare a section with
one list setting named ``fruits``, you can do::

    from sopel.config import types

    class MyPluginSection(types.StaticSection):
        fruits = types.ListAttribute('fruits')

.. note::
    The first ``fruits`` above is the name of this field *on instances of this
    class*, and the second is the name of the section *in the configuration
    file*. It is common (but not required) to use the same name for both.
```

Reply:
This is too much details for a tutorial.